File: js_basic/fizzBuzz.js
const num1 = process.argv[2]
const num2 = process.argv[3]

for (let i = 1; i < 100; i++) {

    if(i % num1 === 0 && i % num2 === 0){

        console.log("fizzbuzz");

    }else if(i % num1 === 0){

        console.log("fizz");

    }else if(i % num2 === 0){

        console.log("buzz");

    }else{

        console.log(i);

    }
 
}

File: ruby-oop/fizzbuzz.rb
class Fizzbuzz
  attr_accessor :num1, :num2

  def initialize(num1, num2)
    @num1 = num1
    @num2 = num2
  end

  def run
    fizzbuzz = []
    for num in 1..100
      str = ""
      str += "fizz" if num % @num1 == 0
      str += "buzz" if num % @num2 == 0
      str += num.to_s if str.length == 0
      fizzbuzz << str
    end
    fizzbuzz
  end
end

fb = Fizzbuzz.new(3, 5)
p fb.run
fb.num1 = 4
fb.num2 = 5
p fb.run
